src/main/python/image_preprocessing/build_director.py

```diff
@@ -17,19 +17,59 @@ class BuildDirector:
     The BuildDirector constructs the Pipeline using the PipelineBuilder
     """
     @staticmethod
-    def construct():
+    def construct(preferences):
         """
         This function constructs the pipeline
         Author(s):
             Nicolai van Niekerk
+        Args:
+            preferences (dict): User-specified techniques to use in pipeline.
         Returns:
             :Pipeline (Constructed pipeline)
         """
         builder = PipelineBuilder()
 
-        blur_manager = BlurManager("median", (9, 9))
-        color_manager = ColorManager("extract", "red")
-        threshold_manager = ThresholdingManager("adaptive")
+        # Use template matching to identify type here
+
+        if 'blur_method' in preferences:
+            blur_method = preferences['blur_method']
+        else:
+            """
+                if type == 'card':
+                    blur_method = 'gaussian'
+                else if type == 'book':
+                    blur_method = 'median'
+            """
+            blur_method = 'median'
+
+        if blur_method == 'median':
+            blur_kernel_size = [9]
+        else:
+            blur_kernel_size = [(3, 3)]
+
+        if 'threshold_method' in preferences:
+            threshold_method = preferences['threshold_method']
+        else:
+            # Default
+            threshold_method = 'adaptive'
+
+        if 'color' in preferences:
```

Comment:
I will add the top line, but the bottom line can't be done because of template matching.